test-magnolia/shared/src/main/scala/zio/test/magnolia/Derivation.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.magnolia
+
+import java.util.UUID
+
+import zio.random.Random
+import zio.test.Gen
+import zio.test.Sized
+
+import magnolia._
+
+object Derivation {
+
+  type Typeclass[T] = Gen[Random with Sized, T]
+
+  implicit val genBoolean: Typeclass[Boolean] = Gen.boolean
+  implicit val genByte: Typeclass[Byte]       = Gen.anyByte
+  implicit val genChar: Typeclass[Char]       = Gen.anyChar
+  implicit val genDouble: Typeclass[Double]   = Gen.anyDouble
+  implicit val genFloat: Typeclass[Float]     = Gen.anyFloat
+  implicit val genInt: Typeclass[Int]         = Gen.anyInt
+  implicit val genLong: Typeclass[Long]       = Gen.anyLong
+  implicit val genShort: Typeclass[Short]     = Gen.anyShort
+  implicit val genString: Typeclass[String]   = Gen.anyString
+  implicit val genUnit: Typeclass[Unit]       = Gen.unit
+  implicit val genUUID: Typeclass[UUID]       = Gen.anyUUID
+
+  implicit def genEither[A, B](implicit ev1: Typeclass[A], ev2: Typeclass[B]): Typeclass[Either[A, B]] =
+    Gen.either(ev1, ev2)
+
+  implicit def genFunction[A, B](implicit ev: Typeclass[B]): Typeclass[A => B] =
+    Gen.function(ev)
+
+  implicit def genList[A](implicit ev: Typeclass[A]): Typeclass[List[A]] =
```

Comment:
Do we need to add instances for `Vector`, etc.?

Reply:
Yes. I missed that one. I added an instance for `Seq` because we need that to derive variable generators for case classes with variable arguments parameters but I think we also need ones for types like `Iterable`. Will take another pass on this.